appcues/src/main/java/com/appcues/ui/AppcuesOverlayViewManager.kt

```diff
@@ -32,19 +32,25 @@ internal class AppcuesOverlayViewManager(private val scope: Scope) : DefaultLife
 
     override fun onResume(owner: LifecycleOwner) {
         super.onResume(owner)
-        viewModel?.onResume()
         shakeGestureListener?.start()
     }
 
     override fun onPause(owner: LifecycleOwner) {
         super.onPause(owner)
-        viewModel?.onPause()
         shakeGestureListener?.stop()
     }
 
-    fun start() {
+    fun start(): Boolean {
+        val activity = AppcuesActivityMonitor.activity
+
+        // inform the caller that we cannot currently add a view in the app
+        if (activity == null || AppcuesActivityMonitor.isPaused) {
```

Comment:
thats an interesting state to safe guard against, the experience showing while the activity is paused would surely cause a problem

Reply:
yes, it essentially leaves the state machine in a hung state - as it cannot progress from BeginningStep to RenderingStep, the UI never moves it forward.